### ISSUE TITLE
Silent failure (crash?) when using `--script <script>` with an invalid script path

### ISSUE THREAD

**Issue by Klaim** [godotengine/godot#96691]
### Tested versions

- v4.3-stable (v4.3.stable.official [77dcf97d8]) : silent failures (unexpected)
- v4.2.2-stable (v4.2.2.stable.official [15073afe3]) : explicit errors (as expected)

### System information

Godot v4.3.stable - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 Ti (NVIDIA; 31.0.15.5222) - AMD Ryzen 9 5900X 12-Core Processor (24 Threads)

### Issue description

When passing an incorrect path to `--script`, v4.3.0 will fail silently, reporting no error.
When doing the same with v4.2.2 we get an explicit error.

### Steps to reproduce

Run (no need to be in a project, same issue whatever the context)
```
godot --headless --script <script>
```
with `<script>` being a path that doesnt lead to any file or resource, or doesnt have an extension.
- v4.3.0 will silently fail
- v4.2.2 will report the problem (either the path leads to no file or it's not the expected extension)

On my system (`git-bash` over `powershell`):
```
$ /e/tools/godot/Godot_v4.2.2-stable_win64.exe --headless --script machin
Godot Engine v4.2.2.stable.official.15073afe3 - https://godotengine.org
 
ERROR: Resource file not found: res://machin (expected type: )
   at: (core/io/resource_loader.cpp:279)
ERROR: Can't load script: machin
   at: (main/main.cpp:3098)

$ /e/tools/godot/Godot_v4.2.2-stable_win64.exe --headless --script machin.gd
Godot Engine v4.2.2.stable.official.15073afe3 - https://godotengine.org
 
ERROR: Attempt to open script 'res://machin.gd' resulted in error 'File not found'.
   at: (modules/gdscript/gdscript.cpp:1025)
ERROR: Failed loading resource: res://machin.gd. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:274)
ERROR: Can't load script: machin.gd
   at: (main/main.cpp:3098)

$ /e/tools/godot/Godot_v4.3-stable_win64.exe --headless --script machin
Godot Engine v4.3.stable.official.77dcf97d8 - https://godotengine.org


$ /e/tools/godot/Godot_v4.3-stable_win64.exe --headless --script machin.gd
Godot Engine v4.3.stable.official.77dcf97d8 - https://godotengine.org

```
Note that for the last 2 commands I see a marker in my bash that shows the program returns non-zero, the value being:
```
$ echo $?
1
```

### Minimal reproduction project (MRP)

N/A

**Comment by AThousandShips** [comment]
Can't replicate the lack of feedback on 4.4 [d0dc3896a], it prints correctly that it can't find the script

I'm not sure what else the engine should do in this case, you're asking to run a script that doesn't exist, other than what program exit code to use

**Comment by Klaim** [comment]
> So not sure what is the problem reported here

The issue is specifically that there are conditions where the errors are not displayed when the execution failed with `v4.3-stable`, but it is visible with `v4.2.2-stable`, however in my initial report I assumed the conditions were specific to the flags passed to Godot, they are not:

As you reported that it works with `v4.4` I tried `v4.4-DEV1` and saw no error output again, so I dug deeper into the issue. I finally found that:
1. the issue is not specific to `--script`, there are many errors that are not reported on exit of Godot at least when running in cli;
2. turns out that the output is different depending on the terminal I'm using:
    - with `git-bash` , `v4.3-stable` and `v4.4-DEV1` 's **error output is not visible** with the command from the report, but `v4.2.2-stable` does report the error (that's what I was seeing in my initial report+various repros as this is my default terminal);
    - with `PowerShell`, **the errors are reported correctly with all versions** but the error output overwrites the normal output (typing anything after the execution of Godot will write over the output) - not sure what's going on there;
    - with `cmd` **the errors are reported correctly with all versions** and the output is correctly displayed;
    
 Because I use `git-bash` by default I couldnt see the error reports and assumed that it was probably Godot crashing when reporting the error, but apparently it's more like an error output routing issue, although I coudlnt check with a debug version of Godot right now, I note to do that when I find the time.
 
So for now I suppose the problem is with `git-bash`, not with Godot, and I'll close this issue. Apparently something changed in `v4.3` and following error output that `git-bash` doesnt digest well.
